--- origsrc/aqualung-0.9beta10/src/plugin.c	2009-02-08 04:55:56.000000000 -0600
+++ src/aqualung-0.9beta10/src/plugin.c	2009-03-04 03:14:51.492250000 -0600
@@ -43,7 +43,7 @@
 #include "plugin.h"
 
 
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__CYGWIN__)
 #define dirent64 dirent
 #define scandir64 scandir
 #define alphasort64 alphasort
@@ -113,8 +113,10 @@ set_active_state(void) {
 static int
 rdf_filter(const struct dirent64 * de) {
 
+#ifdef DT_UNKNOWN
 	if (de->d_type != DT_UNKNOWN && de->d_type != DT_REG && de->d_type != DT_LNK)
 		return 0;
+#endif
 
 	if (de->d_name[0] == '.')
 		return 0;
@@ -127,8 +129,10 @@ rdf_filter(const struct dirent64 * de) {
 static int
 so_filter(const struct dirent64 * de) {
 
+#ifdef DT_UNKNOWN
 	if (de->d_type != DT_UNKNOWN && de->d_type != DT_REG && de->d_type != DT_LNK)
 		return 0;
+#endif
 
 	if (de->d_name[0] == '.')
 		return 0;
